### PR TITLE
Add consistency to imports based on `isort` settings

### PR DESCRIPTION
**Pull request by ThibFrgsGmz**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

This PR consists of de-duplicating, grouping and sorting imports according to isort parameters.

## Rationale

This is to make the code base more consistent, as using a common convention for imports makes the code more readable and idiomatic.

[Ref - Ruff linter](https://beta.ruff.rs/docs/rules/unsorted-imports/)

## Testing/Review Recommendations

Visual inspection

## Future Work

Add Ruff linter into repo's CI (https://github.com/fprime-community/fprime-tools/pull/140).

### CODE REVIEW

File: src/fprime/fbuild/builder.py [hunk 1]
```diff
@@ -7,21 +7,20 @@
 from pathlib import Path
 from typing import Iterable, List, Union
 
+# Forces targets into existence
+import fprime.fbuild.target_definitions  # lgtm[py/unused-import]
```
Comment (thomas-bc) [inline src/fprime/fbuild/builder.py:11]: This import here is quite special and I wonder if we should leave it at the very bottom of the imports, to make it obvious/easily identifiable.

**Review by thomas-bc — COMMENTED** [review]
LGTM. Added a comment for your consideration.

File: src/fprime/fbuild/builder.py (hunk 1 above)
Comment (thomas-bc) [reply src/fprime/fbuild/builder.py:11]: Also, we should probably simply revise that pattern.

Comment (LeStarch) [reply src/fprime/fbuild/builder.py:11]: We agreed this was ok, but revision of the pattern is advisable.

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
@thomas-bc looks good to me.  All yours to close-out.

**Pull request merged**